### PR TITLE
feat(supervision): restart policy and retart params covered

### PR DESCRIPTION
**Pull request by sargarass**
following changes for [PR 118](https://github.com/elfo-rs/elfo/pull/118)

### CODE REVIEW

File: src/ch04-02-supervision.md [hunk 1]
```diff
@@ -30,28 +30,42 @@ More information about configs and the reconfiguration process is available on [
...
-* If there are no restarts for 5s, restart an actor immediately.
-* Otherwise, increase restart time by 5s and schedule restarting.
-* Maximum restart time is limited by 30s.
-
-These constants aren't configurable for now ([elfo#62](https://github.com/elfo-rs/elfo/issues/62)).
 
 The restart policy can be chosen while creating `ActorGroup`:
 ```rust
-use elfo::group::{RestartPolicy, ActorGroup};
+use elfo::group::{RestartPolicy, RestartParams, ActorGroup};
```
Comment (loyd) [inline src/ch04-02-supervision.md:42]: `group::` here is redundant now

File: src/ch04-02-supervision.md [hunk 2]
```diff
@@ -30,28 +30,42 @@ More information about configs and the reconfiguration process is available on [
...
+* `min_backoff` - Minimal restart time limit.
+* `max_backoff` - Maximum restart time limit.
+
+The `RestartParams` can be further configured with the following optional parameters:
+* `factor` - The value to multiply the current delay with for each retry attempt. Default value is `2.0`.
+* `max_retries` - The limit on retry attempts, after which the actor stops attempts to restart. The default value is `NonZeroU64::MAX`, which effectively means unlimited retries.
+* `auto_reset` - The duration of an actor's lifecycle sufficient to deem the actor healthy. The default value is `min_backoff`.
+After this duration elapses, the backoff strategy automatically resets. The following restart will occur **without delays** and will be considered the first retry.
+Subsequent restarts will have delays calculated using the formula above, with the `power` starting from 0.
+
```
Comment (loyd) [inline src/ch04-02-supervision.md:68]: Need to note about `set_restart_policy` and the configuration

Comment (sargarass) [reply src/ch04-02-supervision.md:100]: Added

File: src/ch04-02-supervision.md [hunk 3]
```diff
@@ -28,30 +28,76 @@ The reconfiguration process consists of several stages:
...
 ```
 
+The group restart policy can be overridden separately for each actor group in the configuration:
+```toml
+[group_a]
+[system.restart_policy]
+when = "Never" # Override group policy on `config_update` with `RestartPolicy::never()`.
+
+[group_b]
+[system.restart_policy]
```
Comment (loyd) [inline src/ch04-02-supervision.md:56]: `group_b.system.restart_policy`

File: src/ch04-02-supervision.md [hunk 4]
```diff
@@ -28,30 +28,76 @@ The reconfiguration process consists of several stages:
...
-ActorGroup::new().restart_policy(RestartPolicy::on_failures())
+ActorGroup::new().restart_policy(RestartPolicy::always(RestartParams::new(...)))
+ActorGroup::new().restart_policy(RestartPolicy::on_failure(RestartParams::new(...)))
 ActorGroup::new().restart_policy(RestartPolicy::never())
 ```
 
+The group restart policy can be overridden separately for each actor group in the configuration:
+```toml
+[group_a]
+[system.restart_policy]
```
Comment (loyd) [inline src/ch04-02-supervision.md:52]: `group_a.system.restart_policy`

Comment (sargarass) [reply src/ch04-02-supervision.md]: fixed

File: src/ch04-02-supervision.md (hunk 3 above)
Comment (sargarass) [reply src/ch04-02-supervision.md]: fixed

File: src/ch04-02-supervision.md [hunk 5]
```diff
@@ -28,30 +28,74 @@ The reconfiguration process consists of several stages:
...
+
+// Restore the group policy from the configuration (if specified), or use the default group policy.
+ctx.set_restart_policy(None);
+```
+
+### Repetitive restarts
+Repetitive restarts are limited by an exponential backoff strategy. The delay for each subsequent retry is calculated as follows:
+```
+delay = min(min_backoff * pow(factor, power), max_backoff)
+power = power + 1
```
Comment (loyd) [inline src/ch04-02-supervision.md:82]: I think introducing `power` is useless here. Actually, we're talking about
```
delay = min(min_backoff * pow(factor, restart_count), max_backoff)
          = 0, after resets
```

**Review by loyd — APPROVED** [review]

File: src/ch04-02-supervision.md (hunk 5 above)
Comment (sargarass) [reply src/ch04-02-supervision.md:82]: `restart_count`/`restart_count - 1`  would be confusing here

**Pull request merged**